### PR TITLE
Add com.github.jkotra.unlockr

### PR DESCRIPTION
**Pull request by jkotra**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

---

**password protected file for testing**: [pdf-example-password.pdf](https://github.com/flathub/flathub/files/9152193/pdf-example-password.pdf)
Password: `test`

### CODE REVIEW

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

File: com.github.jkotra.unlockr.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+runtime : org.gnome.Platform
+runtime-version : '42'
+sdk : org.gnome.Sdk
+command: unlockr
+
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:11]: why do you need this?

File: com.github.jkotra.unlockr.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - --filesystem=host
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:12]: that's a very broad filesystem permission

File: com.github.jkotra.unlockr.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - --persist=path
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:13]: not sure this does what you think it does.

File: com.github.jkotra.unlockr.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - --device=dri
+
+modules:
+  - name: jpegsrc
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:17]: libjpeg IS in the runtime already.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
You probably need cleanup rules to remove development files from qpdf.

File: com.github.jkotra.unlockr.yml (hunk 2 above)
Comment (jkotra) [reply com.github.jkotra.unlockr.yml]: I based my manifest on my previous app that required those, didn't notice that. sorry. fixed now.

File: com.github.jkotra.unlockr.yml (hunk 1 above)
Comment (jkotra) [reply com.github.jkotra.unlockr.yml]: according to [docs](https://docs.flatpak.org/en/latest/sandbox-permissions.html#standard-permissions):
`--socket=wayland` for wayland 
`--socket=fallback-x11` if wayland is not available.
`--share=ipc` required by x11.

please suggest if I'm wrong.

**Comment by jkotra** [comment]
removed unnecessary permissions and modules.

---

> You probably need cleanup rules to remove development files from qpdf.

i will look into it.

File: com.github.jkotra.unlockr.yml (hunk 1 above)
Comment (hfiguiere) [reply com.github.jkotra.unlockr.yml]: `- --talk-name=org.freedesktop.Flatpak` is the only one I was asking about here.

Comment (jkotra) [reply com.github.jkotra.unlockr.yml]: removed now.

File: com.github.jkotra.unlockr.yml (hunk 3 above)
Comment (jkotra) [reply com.github.jkotra.unlockr.yml]: removed.

File: com.github.jkotra.unlockr.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+runtime-version : '42'
+sdk : org.gnome.Sdk
+command: unlockr
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --filesystem=xdg-documents
+  - --filesystem=home:ro
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:12]: since the app uses Gtk4, it support portals. It shouldn't even need any permissions.

Comment (nedrichards) [reply com.github.jkotra.unlockr.yml:12]: The home one for sure - the means of operation does need to get permissions on a writable location given that it decrypts and saves the PDF and doesn't want to overwrite the existing encrypted file.

**Comment by nedrichards** [comment]
All seems to work fine, my main concern is that the password is echoed to stdout?

> ** Message: 21:58:43.776: t
** Message: 21:58:43.869: te
** Message: 21:58:44.083: tes
** Message: 21:58:44.307: test

**Comment by jkotra** [comment]
> All seems to work fine, my main concern is that the password is echoed to stdout?
> 
> > ** Message: 21:58:43.776: t
> > ** Message: 21:58:43.869: te
> > ** Message: 21:58:44.083: tes
> > ** Message: 21:58:44.307: test

leftover debug code from development times. its now hidden by default in `0.02`.

File: com.github.jkotra.unlockr.yml (hunk 5 above)
Comment (jkotra) [reply com.github.jkotra.unlockr.yml]: Yes, i removed home but kept `xdg-documents`. PDF gets automatically saved to Documents (overwrite if it already exists)

This is the behaviour for now, might change in future.

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

**Comment by jkotra** [comment]
* might have found a critical bug, please don't merge this yet.

(Sorry for the last minute updates 😞)

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

**Comment by jkotra** [comment]
> You probably need cleanup rules to remove development files from qpdf.

done. size is now reduced to 7.6 mb (from 80mb).

File: com.github.jkotra.unlockr.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+modules:
+  - name: qpdf
+    buildsystem: autotools
+    sources:
+      - type: git
+        url: 'https://github.com/qpdf/qpdf'
+        tag: "release-qpdf-10.6.3"
+    cleanup:
+      - "*.a"
```
Comment (hfiguiere) [inline com.github.jkotra.unlockr.yml:22]: no `/include` ? `*.la` ? pkgconfig files?

Comment (jkotra) [reply com.github.jkotra.unlockr.yml:22]: done. ~600kb reduced.

let me know if i miss anything else. :+1:

**Comment by jkotra** [comment]
bot, build com.github.jkotra.unlockr

**Comment by nedrichards** [comment]
Can you test this build locally? I don't see the test file being saved to documents.

**Comment by jkotra** [comment]
> Can you test this build locally? I don't see the test file being saved to documents.

[tested](https://github.com/flathub/flathub/pull/3343#issuecomment-1191830808) on arch linux and Fedora VM. working as expected.

File: com.github.jkotra.unlockr.yml (hunk 6 above)
Comment (hfiguiere) [reply com.github.jkotra.unlockr.yml:22]: you could check the content of the flatpak. The build dir contain all the files.

(tbf I wish the build bot allowed listing the files)

**Pull request closed without merging**